src/helics/shared_api_library/MessageFederate.h

```diff
@@ -18,95 +19,124 @@ SPDX-License-Identifier: BSD-3-Clause
 extern "C" {
 #endif
 
-/* MESSAGE FEDERATE calls*/
-
-/** create an endpoint
-    @details the endpoint becomes part of the federate and is destroyed when the federate is freed so there are no separate free functions
-    for endpoints
-    @param fed the federate object in which to create an endpoint must have been create with helicsCreateMessageFederate or
-    helicsCreateCombinationFederate
-    @param name the identifier for the endpoint,  this will be prepended with the federate name for the global identifier
-    @param type a string describing the expected type of the publication may be NULL
-    @forcpponly
-    @param[in,out] err a pointer to an error object for catching errors
-    @endforcpponly
-    @return an object containing the endpoint, nullptr on failure
-    */
+/* MessageFederate Calls*/
+
+/**
+ * Create an endpoint.
+ *
+ * @details The endpoint becomes part of the federate and is destroyed when the federate is freed
+ *          so there are no separate free functions for endpoints.
+ *
+ * @param fed The federate object in which to create an endpoint must have been created
+ *           with helicsCreateMessageFederate or helicsCreateCombinationFederate.
+ * @param name The identifier for the endpoint. This will be prepended with the federate name for the global identifier.
+ * @param type A string describing the expected type of the publication may be NULL.
+ * @forcpponly
+ * @param[in,out] err A pointer to an error object for catching errors.
+ * @endforcpponly
+ *
+ * @return An object containing the endpoint, nullptr on failure.
```

Comment:
Do we want to remove the reference to nullptr and reference the isValid functions for checking,  In C it would be NULL but not sure how that translates to the other languages.